contracts/adapters/AaveMigrationExtension.sol

```diff
@@ -0,0 +1,615 @@
+     * @param _amount1Min The minimum amount of token1 that should be accounted for the burned liquidity.
+     */
+    function decreaseLiquidityPosition(
+        uint256 _tokenId,
+        uint128 _liquidity,
+        uint256 _amount0Min,
+        uint256 _amount1Min
+    )
+        external
+        onlyOperator
+    {
+        _decreaseLiquidityPosition(
+            _tokenId,
+            _liquidity,
+            _amount0Min,
+            _amount1Min
+        );
+    }
+
+    /**
+     * @notice OPERATOR ONLY: Migrates a SetToken's position from an unwrapped collateral asset to another SetToken 
+     * that consists solely of Aave's wrapped collateral asset
+     * @param _decodedParams The decoded migration parameters.
+     * @param _underlyingLoanAmount The amount of unwrapped collateral asset to be borrowed via flash loan.
+     * @param _maxSubsidy The maximum amount of unwrapped collateral asset to be transferred to the Extension as a subsidy.
+     * @return underlyingOutputAmount The amount of unwrapped collateral asset returned to the operator.
+     */
+    function migrate(
+        DecodedParams memory _decodedParams,
+        uint256 _underlyingLoanAmount,
+        uint256 _maxSubsidy
+    )
+        external
+        onlyOperator
+        returns (uint256 underlyingOutputAmount)
+    {
+        // Subsidize the migration
+        if (_maxSubsidy > 0) {
+            underlyingToken.transferFrom(msg.sender, address(this), _maxSubsidy);
+        }
+
+        // Encode migration parameters for flash loan callback
+        bytes memory params = abi.encode(_decodedParams);
+
+        // Request flash loan for the underlying token
+        POOL.flashLoanSimple(
+            address(this),
+            address(underlyingToken),
+            _underlyingLoanAmount,
+            params,
+            0
+        );
+
+        // Return remaining underlying token to the operator
+        underlyingOutputAmount = _returnExcessUnderlying();
+    }
+
+    /**
+     * @dev Callback function for Aave V3 flash loan, executed post-loan. It decodes the provided parameters, conducts the migration, and repays the flash loan.
+     * @param asset The asset borrowed in the flash loan.
+     * @param amount The amount borrowed.
+     * @param premium The additional fee charged for the flash loan.
+     * @param initiator The initiator of the flash loan.
+     * @param params Encoded migration parameters.
+     * @return True if the operation is successful.
+     */
+    function executeOperation(
+        address asset,
+        uint256 amount,
+        uint256 premium,
+        address initiator,
+        bytes calldata params
+    )
+        external
+        override
+        returns (bool) 
+    {
+        require(msg.sender == address(POOL), "MigrationExtension: invalid flashloan sender");
+        require(asset == address(underlyingToken), "MigrationExtension: invalid flashloan asset");
```

Comment:
I'd say this check is probably not needed. (Since the tx will fail anyway if you don't flashloan the right asset.
Or does this address any specific security consideration ?

Reply:
removed here [0f92d5a](https://github.com/IndexCoop/index-coop-smart-contracts/pull/162/commits/0f92d5a481850a0e722be83ea8239ccb576ef169)